### ISSUE TITLE
Nano: compilation error

### ISSUE THREAD

**Issue by RolluS** [skycocker/chromebrew#131]
Hello,

First thanks for this awsome package manager!

The problem below can be reproduced on ArnoldTheBat (ATB) i686 and Neverware cloudready (i686 also). Didn't happen on ATB amd64 if I recall correctly, because of precompiled package available.
Dependencies were fine, precompiled or compiled.

Starting from a clean install, compiling nano leads to error :(

> Making all in src
> make[2]: Entering directory `/usr/local/tmp/crew/nano-2.3.6/src'
> gcc -DHAVE_CONFIG_H -I. -I..  -DLOCALEDIR=\"/usr/local/share/locale\" -DSYSCONFDIR=\"/usr/local/etc\" -I/usr/local/include/ncurses  -g -O2 -Wall -MT browser.o -MD -MP -MF .deps/browser.Tpo -c -o browser.o browser.c
> 
> mv -f .deps/browser.Tpo .deps/browser.Po
> gcc -DHAVE_CONFIG_H -I. -I..  -DLOCALEDIR=\"/usr/local/share/locale\" -DSYSCONFDIR=\"/usr/local/etc\" -I/usr/local/include/ncurses  -g -O2 -Wall -MT chars.o -MD -MP -MF .deps/chars.Tpo -c -o chars.o chars.c
> mv -f .deps/chars.Tpo .deps/chars.Po
> gcc -DHAVE_CONFIG_H -I. -I..  -DLOCALEDIR=\"/usr/local/share/locale\" -DSYSCONFDIR=\"/usr/local/etc\" -I/usr/local/include/ncurses  -g -O2 -Wall -MT color.o -MD -MP -MF .deps/color.Tpo -c -o color.o color.c
> color.c: In function ‘color_update’:
> color.c:263:6: error: unknown type name ‘magic_t’
>       magic_t cookie = NULL;
>       ^
> color.c:270:3: warning: implicit declaration of function ‘magic_open’ [-Wimplicit-function-declaration]
>    cookie = magic_open(MAGIC_SYMLINK |
>    ^
> color.c:270:23: error: ‘MAGIC_SYMLINK’ undeclared (first use in this function)
>    cookie = magic_open(MAGIC_SYMLINK |
>                        ^
> color.c:270:23: note: each undeclared identifier is reported only once for each function it appears in
> color.c:274:12: error: ‘MAGIC_ERROR’ undeclared (first use in this function)
>             MAGIC_ERROR);
>             ^
> color.c:275:14: warning: comparison between pointer and integer [enabled by default]
>    if (cookie == NULL || magic_load(cookie, NULL) < 0)
>               ^
> color.c:275:3: warning: implicit declaration of function ‘magic_load’ [-Wimplicit-function-declaration]
>    if (cookie == NULL || magic_load(cookie, NULL) < 0)
>    ^
> color.c:278:7: warning: implicit declaration of function ‘magic_file’ [-Wimplicit-function-declaration]
>        magicstring = magic_file(cookie, openfile->filename);
>        ^
> color.c:278:19: warning: assignment makes pointer from integer without a cast [enabled by default]
>        magicstring = magic_file(cookie, openfile->filename);
>                    ^
> color.c:281:6: warning: implicit declaration of function ‘magic_error’ [-Wimplicit-function-declaration]
>       openfile->filename, magic_error(cookie));
>       ^
> color.c:317:3: warning: implicit declaration of function ‘magic_close’ [-Wimplicit-function-declaration]
>    magic_close(cookie);
>    ^
> make[2]: **\* [color.o] Error 1
> make[2]: Leaving directory `/usr/local/tmp/crew/nano-2.3.6/src'
> make[1]: *** [all-recursive] Error 1
> make[1]: Leaving directory`/usr/local/tmp/crew/nano-2.3.6'
> make: **\* [all] Error 2
> nano failed to install: `make` exited with 2

Any idea please?

Thanks, Reynald

**Comment by Kriskras99** [comment]
Looks like the nano.rb is missing a dependency, libmagic (also known as files).
Download the most recent version at this link: ftp://ftp.astron.com/pub/file/ and build it from source.
Then try installing nano again.

**Comment by RolluS** [comment]
Thank you. I'm a noob at that, any tips to configure, make, make install please? Even a step by step would be awsome

**Comment by Kriskras99** [comment]
Step-by-step:
```
$ cd ~/Downloads
$ wget ftp://ftp.astron.com/pub/file/file-5.25.tar.gz
$ tar -xf file-5.25.tar.gz
$ cd file-5.25
$ ./configure --prefix=/usr/local
$ make
$ sudo make install
$ sudo crew install nano
```
That should work, otherwise just report the errors back.

**Comment by RolluS** [comment]
Thank you ver much.

I was only missing the prefix argument for configure..

I have an error during ./configure:
`configure:3913: ./conftest
./configure: line 3915: ./conftest: Permission denied
configure:3917: $? = 126`

**Comment by Kriskras99** [comment]
There seems to be something wrong with your compiler, try reinstalling buildessential.
If that doesnt work use the following commands and post the output:
```
$ wget https://gist.githubusercontent.com/Kriskras99/57a91030715c8aa65985/raw/ed28680ea7b308fcf7b7758523a11c3ce9f81f96/hello.c
$ gcc -g -o hello hello.c 
```
Good luck

**Comment by RolluS** [comment]
Thanks for your time ;)

````
chronos@localhost ~/Downloads $ gcc -g -o hello hello.c 
chronos@localhost ~/Downloads $ ./hello
bash: ./hello: Permission denied
chronos@localhost ~/Downloads $ sudo ./hello
sudo: unable to execute ./hello: Permission denied
chronos@localhost ~/Downloads $ ls -l hello
-rwxr-xr-x 1 chronos chronos 8036 Jan  2 00:22 hello
chronos@localhost ~/Downloads $ ````

**Comment by RolluS** [comment]
I'm going to clean my installation and will come back.

**Comment by Kriskras99** [comment]
The permission denied is my fault because I forget to say that the home partition is mounted as noexec, sorry. You can temporarily (until the next reboot) mount it as exec with `$ sudo mount -i -o remount,exec /home/chronos/user/`.

**Comment by RolluS** [comment]
OK, now that's OK:

> chronos@localhost ~/Downloads $ ./hello
> Hello Worldchronos@localhost ~/Downloads $

**Comment by Kriskras99** [comment]
And compiling files still gives the same error?

**Comment by RolluS** [comment]
I'm compiling nano again, actually compiling dependencies...

**Comment by RolluS** [comment]
OK, thank you very much Kriskras99.

Here is how we solved my issue:
```
$ sudo mount -i -o remount,exec /home/chronos/user/
$ cd ~/Downloads
$ wget ftp://ftp.astron.com/pub/file/file-5.25.tar.gz
$ tar -xf file-5.25.tar.gz
$ cd file-5.25
$ ./configure --prefix=/usr/local
$ make
$ sudo make install
$ sudo crew install nano
````

Thank you so much.

Now, I will edit my fstab to remove the noexec flag of my home partition. I will edit with... Nano haha many thanks

**Comment by Kriskras99** [comment]
You dont need fstab just reboot

**Comment by Kriskras99** [comment]
And you're welcome

**Comment by RolluS** [comment]
fstab is empty, I then edit mtab. My goal is to remove the noexec flag, and chmod +x if I want in my home folder. because for sure I won't remember, plus /home is not critical as /

**Comment by thedamian** [comment]
Not to just add another drip in the ocean, but I've installed the nano.rb package on a x64 chromebook without issues. Just want to calm anyone that has an x64 machine and wants the incredibly useful nano on their machine.

the the problem is with the 32bit version only???

**Comment by Kriskras99** [comment]
It looks like that, but I don't have a i686 chromebook to test it myself.

**Comment by RolluS** [comment]
I think the problem is on 32 bit, because it didn't happen on ArnoldTheBat amd64 if I recall correctly.
Dependencies and nano were fine, but I think everything was already precompiled.

**Comment by RolluS** [comment]
It haven't happen on ArnoldTheBat amd64 build if I recall correctly, because of precompiled package available, for dependencies and nano.

**Comment by lyxell** [comment]
Duplicate of #103
